servers/physics_3d/joints/jacobian_entry_3d_sw.h

```diff
@@ -163,7 +163,7 @@ class JacobianEntry3DSW {
 	Vector3 m_0MinvJt;
 	Vector3 m_1MinvJt;
 	//Optimization: can be stored in the w/last component of one of the vectors
-	real_t m_Adiag;
+	real_t m_Adiag = 1.0;
```

Comment:
Seems to have a fail condition if it's `<= 0`, maybe the default should be `0` as uninitialized state?
```
servers/physics_3d/joints/jacobian_entry_3d_sw.h
73:             m_Adiag = massInvA + m_0MinvJt.dot(m_aJ) + massInvB + m_1MinvJt.dot(m_bJ);
75:             ERR_FAIL_COND(m_Adiag <= real_t(0.0));
89:             m_Adiag = m_0MinvJt.dot(m_aJ) + m_1MinvJt.dot(m_bJ);
91:             ERR_FAIL_COND(m_Adiag <= real_t(0.0));
104:            m_Adiag = m_0MinvJt.dot(m_aJ) + m_1MinvJt.dot(m_bJ);
106:            ERR_FAIL_COND(m_Adiag <= real_t(0.0));
121:            m_Adiag = massInvA + m_0MinvJt.dot(m_aJ);
123:            ERR_FAIL_COND(m_Adiag <= real_t(0.0));
126:    real_t getDiagonal() const { return m_Adiag; }
166:    real_t m_Adiag;
```

CC @pouleyKetchoupp

Reply:
Yeah it would be good to check uninitialized cases, but we would also need a check in `getDiagonal()` to catch these cases (debug only to avoid potential performance hits in release), something like:
```
real_t getDiagonal() const
{
#ifdef DEBUG_ENABLED
    ERR_FAIL_COND_V(m_Adiag <= real_t(0.0), m_Adiag);
#endif
    return m_Adiag;
}
```

Reply:
Might be worth doing in a separate PR not to grow the scope of this one too much I guess.